.github/workflows/publish-module-index.yml

```diff
@@ -0,0 +1,106 @@
+name: Publish module index
+on:
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  pages: write
+  contents: read
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  upload-index-data:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install packages
+        run: npm ci
+
+      - name: Generate moduleIndex.json
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const script = require("./scripts/github-actions/generate-module-index-data.js")
+            await script({ require, core })
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: moduleIndex.json
+          path: moduleIndex.json
+
+      - name: Log in to Azure
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.PUBLISH_CLIENT_ID }}
+          tenant-id: ${{ secrets.PUBLISH_TENANT_ID }}
+          subscription-id: ${{ secrets.PUBLISH_SUBSCRIPTION_ID }}
+
+      - name: Upload to blob storage
+        uses: azure/CLI@v1
+        with:
+          inlineScript: |
+            az storage blob upload --account-name biceplivedatasaprod --container-name bicep-cdn-live-data-container --name module-index --file moduleIndex.json --auth-mode login --overwrite
```

Comment:
Looks like this step is not new to this PR (just moved from the `update-modules-metadata` workflow), but should we be concerned about a GH action overwriting a manifest file in a bucket named biceplivedatasa**prod**?

Reply:
It will create a new blob. The old blob name `bicepModulesMetadata` was a bit confusing because the file does not contain any metadata, but it's more like a module catalog. I've updated `aka.ms/bicepModulesMetadata` (used in Bicep [here](https://github.com/Azure/bicep/blob/b886719ca38c0b9f7e583919cb59d56a13eeeb36/src/Bicep.LangServer/Completions/PublicRegistryModuleMetadataProvider.cs#L26)) to point to the new blob https://live-data.bicep.azure.com/module-index. Will remove the old blob and update Bicep to use a new aka link `aka.ms/br-modules-index` later.

Reply:
I was more concerned about the fact that a GH action is reaching into something labeled "prod," but since it wasn't introduced in this PR, no need to solve it here.